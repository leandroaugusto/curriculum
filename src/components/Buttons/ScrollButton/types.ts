export interface ScrollButtonProps {
  readonly label: string;
}
