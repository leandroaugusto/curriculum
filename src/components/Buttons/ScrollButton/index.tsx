"use client";

import { ScrollButtonProps } from "./types";

export default function ScrollButton({
  label,
}: ScrollButtonProps): React.ReactElement<HTMLButtonElement> {
  const scrollToBottom: () => void = () => {
    const element = document.getElementById("print");
    const elementTop = element?.offsetTop;
    window.scrollTo({ top: elementTop, behavior: "smooth" });
  };

  return (
    <button data-testid="scroll-to-bottom-button" className="rounded-md bg-slate-500 text-slate-100 px-3 py-1 uppercase text-xs md:text-sm font-bold mt-4 print:hidden" onClick={scrollToBottom}
    >
          {label}
    </button>
  );
}
