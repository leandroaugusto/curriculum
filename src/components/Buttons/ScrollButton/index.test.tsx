import { render, screen, fireEvent } from "@testing-library/react";

import ScrollButton from ".";

global.scrollTo = jest.fn();

describe("Button", () => {
  it("renders the button component", () => {
    const { container } = render(<ScrollButton label="Scroll Button" />);
    expect(container).toMatchSnapshot();
  });

  it("should scroll to  bottom after click the button", () => {
    render(<ScrollButton label="Scroll Button" />);

    const printButton = document.createElement("button");
    printButton.setAttribute("id", "print");
    const buttonTop = printButton.offsetTop;
    document.body.appendChild(printButton);

    const button = screen.getByTestId("scroll-to-bottom-button");

    fireEvent.click(button);

    expect(global.scrollTo).toHaveBeenCalledTimes(1);
    expect(global.scrollTo).toHaveBeenCalledWith({
      top: buttonTop,
      behavior: "smooth",
    });
  });
});
