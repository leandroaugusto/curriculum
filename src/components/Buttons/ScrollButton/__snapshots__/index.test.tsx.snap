// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the button component 1`] = `
<div>
  <button
    class="rounded-md bg-slate-500 text-slate-100 px-3 py-1 uppercase text-xs md:text-sm font-bold mt-4 print:hidden"
    data-testid="scroll-to-bottom-button"
  >
    Scroll Button
  </button>
</div>
`;
